### ISSUE TITLE
GCC 12: cannot compile due to Werror=maybe-uninitialized in jsoncpp

### ISSUE THREAD

**Issue by MayeulC** [LunarG/VulkanTools#1740]
I am trying to build LunarG's Vulkan SDK, and encountering the same issue as that one: https://vulkan.lunarg.com/issue/view/6309e0835df1129602e803eb

The compiler warns that a variable may be uninitialized, and then fails the compilation (`[-Werror=maybe-uninitialized]`) as all warnings are treated as errors.

I'm compilling on vanilla Arch Linux, GCC 12.2.0.

<details><summary>Compilation log></summary>

```
[  6%] Building CXX object via/CMakeFiles/vkvia.dir/__/submodules/jsoncpp/dist/jsoncpp.cpp.o
In file included from /usr/include/c++/12.2.0/bits/new_allocator.h:36,
                 from /usr/include/c++/12.2.0/x86_64-pc-linux-gnu/bits/c++allocator.h:33,
                 from /usr/include/c++/12.2.0/bits/allocator.h:46,
                 from /usr/include/c++/12.2.0/string:41,
                 from /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/json/json.h:127,
                 from /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:76:
In function 'std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = Json::Value::ValueHolder]',
    inlined from 'void Json::Value::swapPayload(Json::Value&)' at /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:2967:12,
    inlined from 'bool Json::Reader::readValue()' at /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:433:31:
/usr/include/c++/12.2.0/bits/move.h:205:7: error: 'v.Json::Value::value_' may be used uninitialized [-Werror=maybe-uninitialized]
  205 |       __a = _GLIBCXX_MOVE(__b);
      |       ^~~
/vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp: In member function 'bool Json::Reader::readValue()':
/vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:432:11: note: 'v' declared here
  432 |     Value v;
      |           ^
In function 'std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = Json::Value::ValueHolder]',
    inlined from 'void Json::Value::swapPayload(Json::Value&)' at /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:2967:12,
    inlined from 'bool Json::Reader::readValue()' at /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:446:33:
/usr/include/c++/12.2.0/bits/move.h:205:7: error: 'v.Json::Value::value_' may be used uninitialized [-Werror=maybe-uninitialized]
  205 |       __a = _GLIBCXX_MOVE(__b);
      |       ^~~
/vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp: In member function 'bool Json::Reader::readValue()':
/vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:445:13: note: 'v' declared here
  445 |       Value v;
      |             ^
In function 'std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = Json::Value::ValueHolder]',
    inlined from 'void Json::Value::swapPayload(Json::Value&)' at /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:2967:12,
    inlined from 'bool Json::OurReader::readValue()' at /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:1355:31:
/usr/include/c++/12.2.0/bits/move.h:205:7: error: 'v.Json::Value::value_' may be used uninitialized [-Werror=maybe-uninitialized]
  205 |       __a = _GLIBCXX_MOVE(__b);
      |       ^~~
/vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp: In member function 'bool Json::OurReader::readValue()':
/vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:1354:11: note: 'v' declared here
 1354 |     Value v;
      |           ^
In function 'std::_Require<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> > std::swap(_Tp&, _Tp&) [with _Tp = Json::Value::ValueHolder]',
    inlined from 'void Json::Value::swapPayload(Json::Value&)' at /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:2967:12,
    inlined from 'bool Json::OurReader::readValue()' at /vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:1392:33:
/usr/include/c++/12.2.0/bits/move.h:205:7: error: 'v.Json::Value::value_' may be used uninitialized [-Werror=maybe-uninitialized]
  205 |       __a = _GLIBCXX_MOVE(__b);
      |       ^~~
/vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp: In member function 'bool Json::OurReader::readValue()':
/vulkansdk/1.3.224.1/source/VulkanTools/submodules/jsoncpp/dist/jsoncpp.cpp:1391:13: note: 'v' declared here
 1391 |       Value v;
      |             ^
cc1plus: all warnings being treated as errors
```

</details>

**Comment by mikes-lunarg** [comment]
I can repro on Arch/GCC 12.2.0

**Comment by mikes-lunarg** [comment]
As mentioned in https://vulkan.lunarg.com/issue/view/6309e0835df1129602e803eb a temporary workaround is to add `-DBUILD_WERROR="OFF"` to the `vulkansdk` build script. If you have already tried building, also remove the directory at `source/VulkanTools/build` for the changes to take effect.
```diff
--- vulkansdk	2022-08-26 07:07:48.000000000 -0700
+++ vulkansdk	2022-09-07 12:03:30.871343401 -0700
@@ -191,32 +191,33 @@
 build_lunarg_tools() {
       echo "~~~Building LunarG VulkanTools~~~"
     cd "$LGVT_DIR"
     [ -d "${LGVT_DIR}/${BUILD_DIR}" ] || mkdir -p "${LGVT_DIR}/${BUILD_DIR}"
     cd "${LGVT_DIR}/${BUILD_DIR}"
     cmake -DCMAKE_BUILD_TYPE="$BUILD_RELDEBINFO_TYPE" \
         -DVULKAN_LOADER_INSTALL_DIR="$ARCHDIR" \
         -DVULKAN_HEADERS_INSTALL_DIR="$ARCHDIR" \
         -DVULKAN_REGISTRY_PATH_HINT="$ARCHDIR" \
         -DVULKAN_VALIDATIONLAYERS_INSTALL_DIR="$ARCHDIR" \
         -DBUILD_TESTS="OFF" \
         -DBUILD_VKTRACEVIEWER="OFF" \
         -DCMAKE_INSTALL_PREFIX="$ARCHDIR" \
         -DCMAKE_INSTALL_LIBDIR="$LIBDIR" \
         -DCMAKE_INSTALL_BINDIR="$BINDIR" \
         -DINSTALL_GTEST="OFF" \
+        -DBUILD_WERROR="OFF" \
         "${LGVT_DIR}"
     make $NUMJOBS install
     cd "${SDKDIR}"
 }
```

**Comment by mikes-lunarg** [comment]
Fixed by #1742
